contracts/resolvers/DelegatableResolverRegistrar.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.4;
+import "./IDelegatableResolverRegistrar.sol";
+import "./DelegatableResolver.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+
+/**
+ * A delegated resolver registrar that allows anyone to register subname
+ */
+contract DelegatableResolverRegistrar is IDelegatableResolverRegistrar, ERC165 {
+    DelegatableResolver public resolver;
+
+    constructor(DelegatableResolver _resolver) {
+        resolver = _resolver;
+    }
+
+    /**
+     * @dev Approve an operator to be able to updated records on a node.
+     * @param name      The encoded subname
+     * @param operator  The address to approve
+     */
+
+    function register(bytes memory name, address operator) external {
+        (bytes32 node, bool authorized) = resolver.getAuthorisedNode(
+            name,
+            0,
+            operator
+        );
+        if (authorized == false) {
```

Comment:
I don't think this does what you want. It won't prevent duplicate subdomains, because it only prevents the same person being issued the same name twice. I think instead you need to keep a mapping of issued names, or modify the DelegatableResolver to keep track of issued names.

Reply:
That's my intention. I only added it to avoiding existing operator to waste the gas. Not sure restricting multiple subname operators at registrar level makes sense because the owner of the resolver contract can always approve multiple operators. If we want to change the delegatable resolver to only have single owner per subname, I think I need to add registrar feature directly in the resolver as @jefflau initially suggested.